CorsixTH/Lua/app.lua

```diff
@@ -1531,15 +1537,48 @@ end
 
 --! Function to check the loaded game is compatible with the program
 --!param save_version (num)
+--!param gfx_set (string) What graphics set is used
 --!return true if compatible, otherwise false
-function App:checkCompatibility(save_version)
+function App:checkCompatibility(save_version, gfx_set)
   local app_version = self.savegame_version
-  if app_version >= save_version or self.config.debug then
+  local err
+
+  -- First check the graphics set matches with the game files
+  if (gfx_set == "demo" and not self.using_demo_files) then
+    err = _S.errors.compatibility_error.demo_in_full
+  elseif (gfx_set == "full" and self.using_demo_files) then
+    err = _S.errors.compatibility_error.full_in_demo
+
+  -- if that's all good, check the save and app version
+  elseif app_version >= save_version or self.config.debug then
     return true
-  else
-    UILoadGame:loadError(_S.errors.compatibility_error)
-    return false
+  else -- savegame newer than application
+    err = _S.errors.compatibility_error.new_in_old
+  end
+
+  UILoadGame:loadError(err)
+  return false
+end
+
+--! Compatibility function to work out the game's graphics set
+--!param state The savegame data
+function App.configureGraphicsSetFlag(state)
+  -- First eliminate all but the first campaign level (same in demo and full)
```

Comment:
We may need an additional comment here that we call the demo "Demo Town" if you load it from Single Scenario menu.

Reply:
I don't see how that's relevant here.